api/src/main/java/org/apache/cloudstack/api/command/user/firewall/CreateFirewallRuleCmd.java

```diff
@@ -254,7 +254,6 @@ public void create() {
                 setEntityUuid(result.getUuid());
             }
         } catch (NetworkRuleConflictException ex) {
-            s_logger.info("Network rule conflict: " + ex.getMessage());
             s_logger.trace("Network Rule Conflict: ", ex);
             throw new ServerApiException(ApiErrorCode.NETWORK_RULE_CONFLICT_ERROR, ex.getMessage());
```

Comment:
The same question here.

Reply:
https://github.com/apache/cloudstack/blob/893a88d225276e45f12f9490e6af2c94a81c2965/api/src/main/java/org/apache/cloudstack/api/ServerApiException.java

I find that their are only two constructors in that Exception class:
public ServerApiException(ApiErrorCode errorCode, String description)
public ServerApiException(ApiErrorCode errorCode, String description, Throwable cause)

Maybe it can be changed to 
throw new ServerApiException(ApiErrorCode.NETWORK_RULE_CONFLICT_ERROR, ex.getMessage(), ex);

Reply:
Ah, I am so sorry. I meant the second constructor. 
What I wanted to say is that it might be better to change the `ex.getMessage` to `ex`. This will maintain the full stack trace.

Reply:
Yes, I agree with you. I'll change it later.